sig/datadog/core/header_collection.rbs

```diff
@@ -0,0 +1,14 @@
+module Datadog
+  module Core
+    class HeaderCollection
+      def get: (::String header_name) -> nil
+      def self.from_hash: (untyped hash) -> HashHeaderCollection
+    end
+
+    class HashHeaderCollection < HeaderCollection
+      def initialize: (untyped hash) -> void
```

Comment:
Minor: I guess the `hash` could be `::Hash`? Although any duck type of `Hash` that provides `each_pair` would work as well I guess :) (I don't know if rbs provides such duck types or not)

Reply:
> any duck type of Hash that provides each_pair would work as well I guess :) (I don't know if rbs provides such duck types or not)

RBs provides interfaces! I have yet to make use of those.